vault-upgrade-hook/src/main/java/biz/netcentric/vlt/upgrade/handler/SlingUtils.java

```diff
@@ -23,12 +26,24 @@
 public class SlingUtils extends OsgiUtil {
 
     public ResourceResolver getResourceResolver(Session session) {
-        try (ServiceWrapper<ResourceResolverFactory> serviceWrapper = getService(ResourceResolverFactory.class);) {
+        try (ServiceWrapper<ResourceResolverFactory> serviceWrapper = getService(ResourceResolverFactory.class)) {
             return serviceWrapper.getService()
                     .getResourceResolver(Collections.<String, Object>singletonMap("user.jcr.session", session));
         } catch (LoginException e) {
             throw new IllegalStateException("Cannot get ResourceResolver from session.", e);
         }
     }
 
+    public boolean hasRunModes(Set<String> requiredRunModes) {
+        return getRunModes().containsAll(requiredRunModes);
```

Comment:
The readme says "To do so set the `runmodes` property to an array of applicable runmodes." Which to me means any runmode in the list is applicable. This implementation means all need to match. So either the documentation should be more clear or the implementation should be changed.
I'd suggest to change the implementation and require only one runmode to match as this will give more flexibility. Or get fancy and allow multiple runmodes in one value joined by a . for example, e.g. "runmodes="[publish1.dev,publish2.test]".

```suggestion
        return !Collections.disjoint(getRunModes(), requiredRunModes);
```

Reply:
In fact I wanted keep it compatible to the way the OsgiInstaller/JcrInstaller looks up artefacts from the repository.

But you are right it will be a pain duplicating `UpgradeAction`s when they are meant to run only on an intersection of runmodes like we do with osgi configuration already. I will adapt the implementation accordingly.

Reply:
IMHO AND use cases are much more likely than OR use cases. I.e. I more often want a config only be applied on instances having all run modes set rather than on instances having any of the given run modes set.

Reply:
Agreed. What do you think about the following to be more precise:

> To do so set the runmodes property to an array of required runmodes. If set the `UpgradeAction`s will only be executed when the environment runs in all of the given runmodes.

Reply:
Maybe more likely, but that case could be covered by using a composite runmode like "authordev". I don't see a workaround for the OR case.

What about the fancy variant, e.g. "runmodes="[publish1.dev,publish2.test]"?

Reply:
This would then be 
```
        Iterator<String> it = requiredRunModes.iterator();
        boolean hasMatchingSetOfRunModes = requiredRunModes.isEmpty();
        while (!hasMatchingSetOfRunModes && it.hasNext()) {
            Collection<String> collectionOfRequiredRunModes = Arrays.asList(it.next().split("\\."));
            hasMatchingSetOfRunModes = getRunModes().containsAll(collectionOfRequiredRunModes);
        }
        return hasMatchingSetOfRunModes;
```